test/cmdlineTests/optimizer_array_sload/input.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.0;
+
+contract Arraysum {
+	uint256[] values;
+
+	function sumArray() public view returns(uint) {
+		uint sum = 0;
+
+		for(uint i = 0; i < values.length; i++)
```

Comment:
Please add a comment: The optimizer should read the length of the array only once.